### PR TITLE
[v2.7] Refactor ETCD snapshot restore test suite

### PR DESCRIPTION
**Pull request by markusewalker**
## Problem

There have been a number of different etcd snapshot restore issues that QA has been seeing for some time. As a result, we have decided to place emphasis on more focused testing efforts in this functional area. However, some of the existing test cases were either outdated, or was not put into a formal test suite.
 
## Solution

This PR makes the `snapshot` package into a formal test suite, similar to how `provisioning`, `deleting` and `nodescaling` is doing. This is the first in a line of PRs to target this effort of more coverage for etcd snapshot and restore.

A new `snapshotInput` needs to be added to the config file. See below on how to use it (also updated in the README):

```
snapshotInput:
  upgradeKubernetesVersion: "v1.26.9-rancher1-1"
  snapshotRestore: "all"
```
 
### Manual Testing

- Locally tested RKE2 and K3S for local backups
- Locally tested K3S for S3 backups

### Automated Testing
Jenkins runs will be provided to reviewers offline.

### CODE REVIEW

File: tests/framework/extensions/ingresses/ingresses.go [hunk 1]
```diff
@@ -19,3 +30,46 @@ func AccessIngressExternally(client *rancher.Client, hostname string, isWithTLS
 
 	return result.Ok, err
 }
+
+// CreateIngress will create an Ingress object in the downstream cluster.
+func CreateIngress(client *v1.Client, ingressName string, ingressTemplate networking.Ingress) (*v1.SteveAPIObject, error) {
+	podClient := client.SteveType(pod)
+	err := kwait.Poll(15*time.Second, 5*time.Minute, func() (done bool, err error) {
```
Comment (thaneunsoo) [inline tests/framework/extensions/ingresses/ingresses.go:37]: Can you use the defaults const that you used in verify.go

File: tests/framework/extensions/provisioning/verify.go [hunk 2]
```diff
@@ -44,6 +44,7 @@ const (
 	hostnameLimit               = 63
 	machineNameAnnotation       = "cluster.x-k8s.io/machine"
 	machineSteveResourceType    = "cluster.x-k8s.io.machine"
```
Comment (thaneunsoo) [inline tests/framework/extensions/provisioning/verify.go:46]: I know this is just been in there but these constants in line 45 and 46 point to the same string, maybe change it to one?

File: tests/framework/extensions/ingresses/ingresses.go (hunk 1 above)
Comment (markusewalker) [reply tests/framework/extensions/ingresses/ingresses.go:37]: Good call out, I can definitely do that

File: tests/framework/extensions/provisioning/verify.go (hunk 2 above)
Comment (thaneunsoo) [reply tests/framework/extensions/provisioning/verify.go:46]: ah missed that part, thanks for clarifying

**Review by thaneunsoo — APPROVED** [review]

**Review by Josh-Diamond — DISMISSED** [review]

**Review by thaneunsoo — APPROVED** [review]

**Review by thaneunsoo — APPROVED** [review]

**Review by Josh-Diamond — DISMISSED** [review]
code review + freeform job results LGTM -- approved ✅

**Review by slickwarren — DISMISSED** [review]

**Review by igomez06 — APPROVED** [review]
LGTM

**Review by slickwarren — APPROVED** [review]

**Review by Josh-Diamond — APPROVED** [review]

**Pull request merged**